md5/src/compress.rs

```diff
@@ -1,165 +1,15 @@
-    b = op_f(b, c, d, a, data[15], RC[15], 22);
-
-    // round 2
-    a = op_g(a, b, c, d, data[1], RC[16], 5);
-    d = op_g(d, a, b, c, data[6], RC[17], 9);
-    c = op_g(c, d, a, b, data[11], RC[18], 14);
-    b = op_g(b, c, d, a, data[0], RC[19], 20);
-
-    a = op_g(a, b, c, d, data[5], RC[20], 5);
-    d = op_g(d, a, b, c, data[10], RC[21], 9);
-    c = op_g(c, d, a, b, data[15], RC[22], 14);
-    b = op_g(b, c, d, a, data[4], RC[23], 20);
-
-    a = op_g(a, b, c, d, data[9], RC[24], 5);
-    d = op_g(d, a, b, c, data[14], RC[25], 9);
-    c = op_g(c, d, a, b, data[3], RC[26], 14);
-    b = op_g(b, c, d, a, data[8], RC[27], 20);
-
-    a = op_g(a, b, c, d, data[13], RC[28], 5);
-    d = op_g(d, a, b, c, data[2], RC[29], 9);
-    c = op_g(c, d, a, b, data[7], RC[30], 14);
-    b = op_g(b, c, d, a, data[12], RC[31], 20);
-
-    // round 3
-    a = op_h(a, b, c, d, data[5], RC[32], 4);
-    d = op_h(d, a, b, c, data[8], RC[33], 11);
-    c = op_h(c, d, a, b, data[11], RC[34], 16);
-    b = op_h(b, c, d, a, data[14], RC[35], 23);
-
-    a = op_h(a, b, c, d, data[1], RC[36], 4);
-    d = op_h(d, a, b, c, data[4], RC[37], 11);
-    c = op_h(c, d, a, b, data[7], RC[38], 16);
-    b = op_h(b, c, d, a, data[10], RC[39], 23);
-
-    a = op_h(a, b, c, d, data[13], RC[40], 4);
-    d = op_h(d, a, b, c, data[0], RC[41], 11);
-    c = op_h(c, d, a, b, data[3], RC[42], 16);
-    b = op_h(b, c, d, a, data[6], RC[43], 23);
-
-    a = op_h(a, b, c, d, data[9], RC[44], 4);
-    d = op_h(d, a, b, c, data[12], RC[45], 11);
-    c = op_h(c, d, a, b, data[15], RC[46], 16);
-    b = op_h(b, c, d, a, data[2], RC[47], 23);
-
-    // round 4
-    a = op_i(a, b, c, d, data[0], RC[48], 6);
-    d = op_i(d, a, b, c, data[7], RC[49], 10);
-    c = op_i(c, d, a, b, data[14], RC[50], 15);
-    b = op_i(b, c, d, a, data[5], RC[51], 21);
-
-    a = op_i(a, b, c, d, data[12], RC[52], 6);
-    d = op_i(d, a, b, c, data[3], RC[53], 10);
-    c = op_i(c, d, a, b, data[10], RC[54], 15);
-    b = op_i(b, c, d, a, data[1], RC[55], 21);
-
-    a = op_i(a, b, c, d, data[8], RC[56], 6);
-    d = op_i(d, a, b, c, data[15], RC[57], 10);
-    c = op_i(c, d, a, b, data[6], RC[58], 15);
-    b = op_i(b, c, d, a, data[13], RC[59], 21);
-
-    a = op_i(a, b, c, d, data[4], RC[60], 6);
-    d = op_i(d, a, b, c, data[11], RC[61], 10);
-    c = op_i(c, d, a, b, data[2], RC[62], 15);
-    b = op_i(b, c, d, a, data[9], RC[63], 21);
-
-    state[0] = state[0].wrapping_add(a);
-    state[1] = state[1].wrapping_add(b);
-    state[2] = state[2].wrapping_add(c);
-    state[3] = state[3].wrapping_add(d);
-}
-
-#[inline]
-pub fn compress(state: &mut [u32; 4], blocks: &[[u8; 64]]) {
-    for block in blocks {
-        compress_block(state, block)
+cfg_if::cfg_if! {
+    if #[cfg(feature = "force-soft")] {
+        mod soft;
+        pub use soft::compress;
+    } else if #[cfg(all(feature = "asm", any(target_arch = "x86", target_arch = "x86_64")))] {
+        extern crate md5_asm;
```

Comment:
We don't need to use `extern crate` here. The current code uses it for renaming purposes.